tools/flang1/flang1exe/semfunc.c

```diff
@@ -7411,7 +7410,8 @@ ref_pd(SST *stktop, ITEM *list)
       XFR_ARGAST(3);
       dtype2 = SST_DTYPEG(stkp);
       if (!DT_ISINT(DTY(dtype2 + 1)) ||
-          count != get_int_cval(sym_of_ast(AD_NUMELM(AD_DPTR(dtype2))))) {
+          ((STYPEG(sym_of_ast(AD_NUMELM(AD_DPTR(dtype2)))) == ST_CONST) && 
```

Comment:
Looks like this line of code fixes this test https://github.com/flang-compiler/flang/commit/e3dbec73c851744f7075d24d4d6fef634c9c391d.

Could you explain what other fixes do? Is there an existing issue other fixes are targeting at? Thanks.

Reply:
@shivaramaarao @pawosm-arm Can you shed some light on this? IIUC this PR was supposed to fix the two test cases in 8d1042ea9a797215f35374acb7375fe9d288835f, but that commit was later reverted and only one of the test cases has remained. Is there a reproducer for the other problem fixed by this PR?

Reply:
@bryanpkc  I will get the reproducer testcases for these issues. I assumed that they were present.

Reply:
Thanks @shivaramaarao!